xmr-btc/tests/e2e.rs

```diff
@@ -400,4 +403,126 @@ mod tests {
             initial_balances.bob_btc - swap_amounts.btc - lock_tx_bitcoin_fee
         );
     }
+
+    #[tokio::test]
+    async fn recover_protocol_state_from_db() {
```

Comment:
It wrote it to help me iterate on the DB implementation without having to spin up containers. If anyone wants to modify the DB code and test it quickly, it would help them to. I am also testing the case when I insert twice to the same key. I had a bug earlier where the db would break if I saved to the same key but it worked with an empty db.

Reply:
> It wrote it to help me iterate on the DB implementation without having to spin up containers

But this test does spin up the containers, which I don't think you need to test the db. I think the storage is completely decoupled from the protocol execution. 

It would be a different story if you were testing the application and you wanted to verify that the data was being saved during execution.

Reply:
Oh right my bad, I mistook this test for the one in the storage module. Yes you are correct, I think this test can be removed.

Reply:
> 🔧 I think this test is not adding much. The storage APIs are already tested in the `storage` module. And the protocol is already tested in the other tests.

From the name of the function, I would tend to disagree with you as I just discovered a bug in nectar that would reset some data everytime we load the db. reloading the db and expecting same data is a useful test I believe.

But maybe this could just be a test in storage that does not do the full execution of the swap.